Add milestone to PR by name instead of number

venyii opened the pull request:
Now it's possible to add a milestone by name to a pull request instead of looking up the number. If the milestone can't be found it'll stop before creating the PR.

Closes #1089

mislav commented on features/pull_request.feature:
```diff
@@ -757,19 +757,44 @@ BODY
+        json [
+          { :number => 1234, :title => "prerelease" },
+          { :number => 1337, :title => "v1" },
+          { :number => 41319, :title => "Hello World!" }
```
You can return an empty JSON array here, since we know in advance that no milestones will match "unkown".

mislav commented on commands/pull_request.go:
```diff
@@ -423,3 +428,16 @@ func parsePullRequestIssueNumber(url string) string {
 	return ""
 }
+
+func fetchMilestoneNumber(client *github.Client, project *github.Project, name string) (int, error) {
```
To simplify this method's signature, instead of passing in `client` and `project` to this method, how about passing an already fetched list of milestones?

mislav commented on commands/pull_request.go:
```diff
@@ -423,3 +428,16 @@ func parsePullRequestIssueNumber(url string) string {
+	utils.Check(err)
+
+	for _, milestone := range milestones {
+		if milestone.Title == name {
```
How about case-insensitive match?

venyii commented:
@mislav thanks for the quick review, I implemented your comments - ready for review again :)

venyii commented:
Yes, thought about this as well but went ahead with it since it was in the 2.3 project 😏.

Off the top of my head, we could use the old behaviour **if** the number is an integer (successful `strconv.Atoi`)? That would of course prohibit the use of milestone names that are just integers, but would maybe be enough BC until 3.0 comes around (if planned)?

I think a completely new option isn't worth it and produces BC breaks anyway in the next major version if we intend to replace the old `-M` behavior.

Just speaking for myself, I used the option once and didn't bother with it again since it was too feasible for me to search the milestone number every time 😃 but not sure how others use this.

Either way, if you've got another idea I've got no problems changing the PR! Just let me know and I'll make the changes

venyii commented:
ping @mislav, could you take a look at my last comment regarding the BC? I'd love to get this merged :)

mislav commented:
> Off the top of my head, we could use the old behaviour **if** the number is an integer (successful `strconv.Atoi`)? That would of course prohibit the use of milestone names that are just integers, but would maybe be enough BC until 3.0 comes around (if planned)?

Yes please; let's check if the number looks like an integer and use it as milestone ID for backwards compatibility.

venyii commented:
@mislav done! It now uses the given number directly if it passes the integer conversion, also added a test for it. I added the deprecation info to the parameter doc, let me know if that's enough info

mislav commented:
Thank you a ton!! 🏆
